src/Servers/HttpSys/test/FunctionalTests/Listener/RequestHeaderTests.cs

```diff
@@ -43,14 +44,83 @@ public async Task RequestHeaders_ClientSendsUtf8Headers_Success()
+                Assert.Equal(new[] { new Uri(address).Authority }, requestHeaders.GetValues("Host"));
+                Assert.Equal("close", requestHeaders["Connection"]);
+                Assert.Equal(new[] { "close" }, requestHeaders.GetValues("Connection"));
+                // Apparently Http.Sys squashes request headers together.
+                Assert.Equal("�", requestHeaders["Custom-Header"]);
+                Assert.Equal(new[] { "�" }, requestHeaders.GetValues("Custom-Header"));
+                Assert.Equal("spacervalue", requestHeaders["Spacer-Header"]);
+                Assert.Equal(new[] { "spacervalue" }, requestHeaders.GetValues("Spacer-Header"));
+                context.Dispose();
+
+                await responseTask;
+            }
+        }
+
+        [ConditionalFact]
+        public async Task RequestHeaders_ClientSendsLatin1Headers_Success()
+        {
+            string address;
+            using (var server = Utilities.CreateHttpServer(out address))
+            {
+                server.Options.UseLatin1RequestHeaders = true;
+                string[] customValues = new string[] { "£" };
+                Task responseTask = SendRequestAsync(address, "Custom-Header", customValues, Encoding.Latin1);
+
+                var context = await server.AcceptAsync(Utilities.DefaultTimeout);
+                var requestHeaders = context.Request.Headers;
+                Assert.Equal(4, requestHeaders.Count);
+                Assert.Equal(new Uri(address).Authority, requestHeaders["Host"]);
+                Assert.Equal(new[] { new Uri(address).Authority }, requestHeaders.GetValues("Host"));
+                Assert.Equal("close", requestHeaders["Connection"]);
+                Assert.Equal(new[] { "close" }, requestHeaders.GetValues("Connection"));
+                // Apparently Http.Sys squashes request headers together.
+                Assert.Equal("£", requestHeaders["Custom-Header"]);
+                Assert.Equal(new[] { "£" }, requestHeaders.GetValues("Custom-Header"));
+                Assert.Equal("spacervalue", requestHeaders["Spacer-Header"]);
+                Assert.Equal(new[] { "spacervalue" }, requestHeaders.GetValues("Spacer-Header"));
+                context.Dispose();
+
+                await responseTask;
+            }
+        }
+
+        [ConditionalFact]
+        public async Task RequestHeaders_ClientSendsBadLatin1Headers_Rejected()
+        {
+            string address;
+            using (var server = Utilities.CreateHttpServer(out address))
+            {
+                server.Options.UseLatin1RequestHeaders = true;
+                string[] customValues = new string[] { "£\0a" };
+                var responseTask = SendRequestAsync(address, "Custom-Header", customValues, Encoding.Latin1);
+                var response = await responseTask;
+                Assert.StartsWith("400", response.Substring(9));
+            }
+        }
```

Comment:
@wtgodbe : Is this the behavior on IIS too?

Reply:
Yes, same behavior there. Here's the equivalent test in IIS: https://github.com/dotnet/aspnetcore/blob/e034ff60a86b9ad28a4e8c558592ca302cd63fea/src/Servers/IIS/IIS/test/Common.FunctionalTests/Latin1Tests.cs#L78

Reply:
Thank you!